api/ingress/v1alpha1/ngrok_common.go

```diff
@@ -463,6 +463,14 @@ func (amazon *EndpointOAuthAmazon) ToNgrok(clientSecret *string) *ngrok.Endpoint
 	return mod
 }
 
+// EndpointTrafficPolicy is not exposed and is mostly used for assisting in processing Gateway API filters.
+type EndpointTrafficPolicy struct {
+	// OnHttpRequest traffic rule
+	OnHttpRequest []EndpointRule `json:"on_http_request,omitempty"`
+	// OnHttpResponse traffic rule
+	OnHttpResponse []EndpointRule `json:"on_http_response,omitempty"`
```

Comment:
Nit: In k8s we prefer json fields be `camelCased` and the ngrok-go-sdk prefers them `lower_snake_cased`

@jonstacks can say more to this

Reply:
this is going to get written to the `<protocol>Edge` CRDs, then directly extracted and shoved into the ngrok API.   so we need it in the format we are eventually going to send it in. 

we could add additional code in the reconciler to convert between camel and snake case, but i feel like i'd rather not do that. especially because, with the `NgrokTrafficPolicy` CRD, the phases will be camel case anyway. So this will make the `NgrokTrafficPolicy` CRD look the same between the different configuration methods 🤷

Reply:
> we could add additional code in the reconciler to convert between camel and snake case

we have some special top-level key processing already in the reconciler, but that stuff is theoretically temporary and would be _really_ nice to rip out eventually

Reply:
i know that's a lot of stuff, please let me know if you want me to elaborate on anything here